### DISCUSSION TITLE
How to debug RGT not working

### DISCUSSION THREAD

**Discussion by danbruno** [cagnulein/qdomyos-zwift#909] [Q&A]
I managed to get Zwift working on my Echelon tablet. It auto-connected to a Wahoo Kickr 0000 sensor which worked perfectly. When I tried to do the same with RGT, it seems to connect, but everything is 0.

If I try to connect to the name of the device, it never seems to work. I tried the same by installing it to my Samsung S10 and no luck there either.

What kind of settings can I play around with to try to get it working? I can grab the debug logs if needed, but given that it's working with Zwift, I'm not sure what's going on.

**Comment by cagnulein** [comment]
@danbruno if it's working on zwift, qz is configured correctly. The best way to figure it out it's write to RGT Support and put me in CC explaining that your setup works with zwift
let me know

**Reply by danbruno** [reply]
Appreciate it, I've reached out to support through their ticketing system. Hopefully they can help out.

**Reply by cagnulein** [reply]
ok let me know @danbruno

**Reply by danbruno** [reply]
So far nothing, they said the app wasn't supported, but I pushed if they could provide me with any more information to help me debug why mine isn't working since other people have been able to do it.

I did manage to find their log file which contains FTMS data. Here's one such example. I don't know much about the protocol and the spec is gigantic. Does anything look off here? I replied to the ticket with the same info so they can take a look too. 

```
2022-08-23 10:21:15 PM
RGT: FTMS indicate 64 2 B0 4 40 0 1 0 5 0 0 0 1 6 0 0 0 18 0 0 2A 63 0 0 10 0 80 0 0 80 5F 9B 34 FB 20 0 5 0 20 0 36 D7 1 6 0 0 0 12 0 0 2A 37 0 0 10 0 80 0 0 80 5F 9B 34 FB 0 0 1 6 0 0 0 15 0 0 2A 5B 0 0 10 0 80 0 0 80 5F 9B 34 FB 2 20 0 36 D7 1 6 0 0 0 18 0 0 2A CC 0 0 10 0 80 0 0 80 5F 9B 34 FB 83 14 0 0 C E0 0 0 

Tue Aug 23 22:21:16 2022 1661307676969 Debug:   "Wahoo KICKR 0000" sending to "::ffff:192.168.0.189" : 60719  notification for uuid =  "2ad2" rv= true "64 02 b0 04 40 00 01 00 05 00 00 00"
Tue Aug 23 22:21:16 2022 1661307676971 Debug:   "Wahoo KICKR 0000" sending to "::ffff:192.168.0.189" : 60719  notification for uuid =  "2a63" rv= true "20 00 05 00 20 00 36 d7"
Tue Aug 23 22:21:16 2022 1661307676973 Debug:   "Wahoo KICKR 0000" sending to "::ffff:192.168.0.189" : 60719  notification for uuid =  "2a37" rv= true "00 00"
Tue Aug 23 22:21:16 2022 1661307676976 Debug:   "Wahoo KICKR 0000" sending to "::ffff:192.168.0.189" : 60719  notification for uuid =  "2a5b" rv= true "02 20 00 36 d7"
Tue Aug 23 22:21:16 2022 1661307676978 Debug:   "Wahoo KICKR 0000" sending to "::ffff:192.168.0.189" : 60719  notification for uuid =  "2acc" rv= true "83 14 00 00 0c e0 00 00"
```

**Comment by cagnulein** [comment]
@danbruno could you post a QZ vs RGT log using bluetooth instead of wifi? Maybe append the full RGT log so I can check it
Let me know

**Reply by danbruno** [reply]
I actually managed to get it working this morning with my phone only. Seems that if the phone is bluetooth paired to my laptop, it won't connect over bluetooth FTMS. Makes sense in hindsight, but I forgot it was even paired. I was hoping to use the tablet the bike came with, that seems a bit more spotty. Sometimes it doesn't connect at all and other times connects but seems to send nothing. The problem is that the RGT debug logs over bluetooth only seem to have connect/disconnect logs. I don't see any details of actual transmissions at all.

I'll grab my tablet vs S10 logs after work today and see if there's anything I can find that's different between them.

Appreciate all the help so far, at least I can get it working through my phone in the worst case scenario.

**Reply by cagnulein** [reply]
great! sure let me know how this will go

**Reply by danbruno** [reply]
I tried on both and for some reason I can't find the logs for my S10, it's not in the same place as my tablet and searching for it finds nothing.

That said, it seems the tablet is getting constant authorization issues. Have you ever seen this before? Here's a link to the [logs](https://docs.google.com/document/d/1zAsbZar90sIC-G7rl06Fx2Qf3XqjieyAaJxqeTwjNoM/edit?usp=sharing)

I've run nRF Connect on my S10 to try to connect to the tablet to see what it looks like. Do I need to bond to it, or is connecting sufficient? Bonding seems to fail, but the connection itself works. 

While reading the characteristics, it the name first "rk3288", then 0x00-00 from 00002a01-0000-1000-8000-00805f9b34fb, then "Address resolution not supported" 0x00 from 00002aa6-0000-1000-8000-00805f9b34fb

Followed by what appears like nothing for 00002a38-0000-1000-8000-00805f9b34fb, then nothing from 0000aaa1-0000-1000-8000-aabbccddeeff followed by a GATT_ERROR and a disconnection 1ms after reading from the last characteristic.

Not sure if that helps or not.

It reads a few connections and then gets Error 133 GATT ERROR while

**Reply by cagnulein** [reply]
> I tried on both and for some reason I can't find the logs for my S10, it's not in the same place as my tablet and searching for it finds nothing.

are you running android 13 there? are you using cx file explorer?

> That said, it seems the tablet is getting constant authorization issues. Have you ever seen this before? Here's a link to the [logs](https://docs.google.com/document/d/1zAsbZar90sIC-G7rl06Fx2Qf3XqjieyAaJxqeTwjNoM/edit?usp=sharing)

ignore that error, it's just the last error happened. the communication to your bike is working well (i see the metrics correctly)

> I've run nRF Connect on my S10 to try to connect to the tablet to see what it looks like. Do I need to bond to it, or is connecting sufficient? Bonding seems to fail, but the connection itself works.

no, never bond it.

> While reading the characteristics, it the name first "rk3288", then 0x00-00 from 00002a01-0000-1000-8000-00805f9b34fb, then "Address resolution not supported" 0x00 from 00002aa6-0000-1000-8000-00805f9b34fb

rename the s10 to "bike" and reboot it, it should help the bt compatibility. those characteristics are standard chars from the phone itself, you should look to 0x1826 service only

let me know

**Reply by danbruno** [reply]
Thanks! I did find the logs on my S10 and honestly couldn't find an appreciable difference between the two logs aside from that random authorization issue. It looks like it's sending but isn't being received for some reason. I'll probably just use my phone in the meantime.

Appreciate the help though. RGT is aware that they do have some ftms issues from some devices that hopefully they will be able to work through.

**Reply by cagnulein** [reply]
ok thanks @danbruno!
